sklearn/ensemble/forest.py

```diff
@@ -354,6 +384,17 @@ def _validate_X_predict(self, X):
 
         return self.estimators_[0]._validate_X_predict(X, check_input=True)
 
+    def _validate_missing_mask(self, X, missing_mask=None):
```

Comment:
Without missing mask [this](https://github.com/scikit-learn/scikit-learn/pull/5974/files#diff-23467bb90c859a3e381a82626abe64d6R779) would have to be `isnan`. And the reason why we decided to go with `missing_mask` was that nan representations differ and hence making the comparison of float to `nan` costly. (Ref: https://github.com/scikit-learn/scikit-learn/issues/5870#issuecomment-159409438)